### PR TITLE
Refactor the code to separate out the store package 

### PR DESCRIPTION
**Pull request by soumya-codes**
This PR is the first major in the many set of refactoring PRs required for moving DiceDB to multithreading.

This PR refactors the existing code by doing the following:

1.  Carve out the store code into a separate package under /internal.
2. Move regex to an independent package under /internal
3. Move diceerrors from /core to /internal

**Note**: There is lot of movement of code and takes a lot of effort to rebase and verify changes in this PR so there are couple of requests from my side:
1. Please `DO NOT` merge any further PRs till this PR is merged.
2. Please review the code changes thoroughly.

cc @JyotinderSingh @lucifercr07 @AshwinKul28 @pratikpandey21

### CODE REVIEW

**Comment by AshwinKul28** [comment]
Going through it, just a general comment, can we make `internal/diceerrors` folder rename it to `internal/errors`. It doesn't make sense of having dice as a prefix to any of these internal folders. You can make the change directly by replacing it now. 
@soumya-codes

**Comment by soumya-codes** [comment]
> can we make `internal/diceerrors` folder
 Can I do that in the subsequent PR? There are a lot of areas of improvement here..

File: server/async_tcp.go [hunk 1]
```diff
@@ -224,12 +227,8 @@ func (s *AsyncServer) eventLoop(ctx context.Context) error {
 						if errors.Is(err, ErrAborted) {
 							log.Info("Received abort command, initiating graceful shutdown")
 							return err
-						} else if errors.Is(err, syscall.ECONNRESET) || errors.Is(err, net.ErrClosed) {
-							// Both are normal scenarios when client disconnections happen, hence just info log
-							log.Info("Connection closed by client or reset", "fd", event.Fd)
-						} else {
-							log.Warn(err)
```
Comment (JyotinderSingh) [inline server/async_tcp.go:231]: Let's retain this.

**Review by JyotinderSingh — CHANGES_REQUESTED** [review]
Thanks for this massive effort @soumya-codes! The changes look really well thought out.
Have left one minor comment.

File: server/async_tcp.go (hunk 1 above)
Comment (soumya-codes) [reply server/async_tcp.go:231]: My bad... Got missed out during the merge!

**Review by JyotinderSingh — APPROVED** [review]
LGTM

**Pull request merged**